### ISSUE TITLE
stuck on Indexing database

### ISSUE THREAD

**Issue by treasuretron** [ssbc/patchwork#1223]
Tried to just post this in ssb but got the "Cowardly refusing to publish your message while database is still indexing. Please try again once indexing is finished."

Patchwork seems stuck on Indexing database. Been left for cumulative hours, doesn't seem to be using a ton of CPU. Macbook 12". No blobs seem to be loading. Updated to to 3.17.1 (3.17.1.675) but didn't seem to help. Macos 10.14.6.

Thanks! ;D

**Comment by christianbundy** [comment]
@treasuretron 

Thanks for the bug report! Sorry you're bumping into this, I think someone else mentioned the same problem. Does restarting help? And if not, could you try starting Patchwork from a terminal to see if there's any error output? Feel free to ping me on Signal if you'd like a hand with this.

**Comment by treasuretron** [comment]
thanks @christianbundy. Not sure if you meant restarting my machine or restarting Patchwork but neither make a difference.

Running `ssb-server start` gives me a loooong output that starts with:

`(node:709) UnhandledPromiseRejectionWarning: OpenError: IO error: lock /Users/trav/.ssb/blobs_push/LOCK: Resource temporarily unavailable
    at /usr/local/lib/node_modules/ssb-server/node_modules/levelup/lib/levelup.js:87:23
    at /usr/local/lib/node_modules/ssb-server/node_modules/abstract-leveldown/abstract-leveldown.js:30:14
    at /usr/local/lib/node_modules/ssb-server/node_modules/deferred-leveldown/deferred-leveldown.js:20:21
    at /usr/local/lib/node_modules/ssb-server/node_modules/abstract-leveldown/abstract-leveldown.js:30:14
    at /usr/local/lib/node_modules/ssb-server/node_modules/abstract-leveldown/abstract-leveldown.js:30:14
(node:709) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:709) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:709) UnhandledPromiseRejectionWarning: OpenError: IO error: lock /Users/trav/.ssb/db/LOCK: Resource temporarily unavailable
    at /usr/local/lib/node_modules/ssb-server/node_modules/levelup/lib/levelup.js:87:23
    at /usr/local/lib/node_modules/ssb-server/node_modules/abstract-leveldown/abstract-leveldown.js:30:14
    at /usr/local/lib/node_modules/ssb-server/node_modules/deferred-leveldown/deferred-leveldown.js:20:21
    at /usr/local/lib/node_modules/ssb-server/node_modules/abstract-leveldown/abstract-leveldown.js:30:14
    at /usr/local/lib/node_modules/ssb-server/node_modules/abstract-leveldown/abstract-leveldown.js:30:14
(node:709) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
ssb-friends: stream legacy api used
`
and is a whole bunch of that for a while and ends with
`  at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:48:11
  Error: past end:765430172-2784160752 < 807811345
    at Object.read (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:49:17)
    at next (/usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:47:16)
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:43:11
    at /usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:96:11
    at get (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:29:7)
    at Object.readUInt32BE (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:93:9)
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:41:16
    at /usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:96:11
    at get (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:29:7)
    at Object.readUInt32BE (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:93:9)
Error: view stream error, from:links2
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumedb/index.js:139:35
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-write/index.js:40:31
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-write/index.js:41:13
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-stream/throughs/async-map.js:28:19
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-stream/throughs/filter.js:17:11
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-cursor/skip.js:12:26
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-cursor/cursor.js:28:28
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/inject.js:43:22
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:48:11
  Error: past end:765430172-2784160752 < 807811345
    at Object.read (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:49:17)
    at next (/usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:47:16)
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:43:11
    at /usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:96:11
    at get (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:29:7)
    at Object.readUInt32BE (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:93:9)
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:41:16
    at /usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:96:11
    at get (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:29:7)
    at Object.readUInt32BE (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:93:9)
Error: view stream error, from:links2
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumedb/index.js:139:35
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-write/index.js:40:31
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-write/index.js:41:13
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-stream/throughs/async-map.js:28:19
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-stream/throughs/filter.js:17:11
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-cursor/skip.js:12:26
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-cursor/cursor.js:28:28
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/inject.js:43:22
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:48:11
  Error: past end:765430172-2784160752 < 807811345
    at Object.read (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:49:17)
    at next (/usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:47:16)
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:43:11
    at /usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:96:11
    at get (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:29:7)
    at Object.readUInt32BE (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:93:9)
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/frame/recoverable.js:41:16
    at /usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:96:11
    at get (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:29:7)
    at Object.readUInt32BE (/usr/local/lib/node_modules/ssb-server/node_modules/aligned-block-file/blocks.js:93:9)
/usr/local/lib/node_modules/ssb-server/node_modules/explain-error/index.js:23
  var stack = removePrefix(getStack(_err).slice(1), getStack(err)).join('\n')
                                                                   ^

RangeError: Maximum call stack size exceeded
    at Array.join (native)
    at module.exports (/usr/local/lib/node_modules/ssb-server/node_modules/explain-error/index.js:23:68)
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumedb/index.js:139:35
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-write/index.js:40:31
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-write/index.js:41:13
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-stream/throughs/async-map.js:28:19
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-stream/throughs/filter.js:17:11
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-cursor/skip.js:12:26
    at /usr/local/lib/node_modules/ssb-server/node_modules/pull-cursor/cursor.js:28:28
    at /usr/local/lib/node_modules/ssb-server/node_modules/flumelog-offset/inject.js:43:22
`
I guess I could throw the whole output into pastebin if that's helpful? Thanks for responding!

**Comment by christianbundy** [comment]
Could you try deleting your indexes? Here's a quick set of instructions: https://github.com/ssbc/patchwork/issues/1192#issuecomment-540863309

Note: you can skip the "fix the log" step, I don't think that's relevant to your problem.

**Comment by treasuretron** [comment]
brilliant 🙏 thank you @christianbundy that did the trick!
